CodeMaid/Logic/Cleaning/FileHeaderLogic.cs

```diff
@@ -58,6 +61,8 @@ internal void UpdateFileHeader(TextDocument textDocument)
                 return;
             }
 
+            settingsFileHeader = ReplaceVariables(textDocument, settingsFileHeader);
+
             var cursor = textDocument.StartPoint.CreateEditPoint();
             var existingFileHeader = cursor.GetText(settingsFileHeader.Length);
```

Comment:
Is it safe to assume that if a file starts with a comment it is probably a header and we don't change anything ?

Reply:
That's a good shortcut, and you could simplify it further to just look for the same few starting characters since some languages have multiple possible comment prefixes (e.g. `//` or `/*`).  However anytime someone renames a file or changes the header content, the header is going to out of date and won't be updated.

The old static approach simply inserted the new header anytime it was not already present as expected.  It had the draw back that if someone changed the file header, all files would get the new header and the old headers would have to be manually deleted (e.g. find/replace).  If we stuck with that approach then the new correct header would be inserted over the top of the old outdated header.  I don't think it's the greatest experience, but it's predictable and we definitely want to avoid any scenario of trying to auto-delete an old header (better to leave the code twice then risk losing it).

Thoughts on just going back to the previous logic of inserting whenever the calculated header isn't already present?

Reply:
Yep the problem I see with the old method is when we add other variables that varies like user or time. That won't work with the old static approach. I think if someone renames a file and he want the header to be updated, he just has to delete the header... not that big of a deal

Reply:
Yep the problem I see with the old method is when we add other variables that varies like username or date of creation. That won't work with the old static approach. I think if someone renames a file and he want the header to be updated, he just has to delete the header... not that big of a deal

Reply:
Yeah, agree.  So will you please revert the logic below and go back to just checking for an exact match?  Then you can delete the two GetCodeLanguageComment methods as well (the less code the better).

Reply:
If you revert to the old method, things like name or date variable will always get replaced, I think the new method is better

Reply:
Ok, I'm willing to go with that.

I'm still not a fan of the hard-coded comment prefix list.  The file headers can be any kind of content (e.g. `using System;` or `#pragma`) so it would infinitely re-add if anything _except_ the expected comment prefix was present.  Additionally the `string.Empty` cases are always going to match since `str.StartsWith(string.Empty)` always returns true.

I think an alternative (and perhaps nice compromise) solution would be to still use the expected file header as the comparison point, but cut-off at the first dynamic element.  In other words before ReplaceVariables is called, create a method called GetStaticPrefix that would iterate across the variables, find the lowest IndexOf the variables and return the substring that matches up to that (defaulting to the full string when no variables are present).  Make sense?

Reply:
Yep, I'll add that, thanks